app/navigation/root-navigator.tsx

```diff
@@ -271,18 +270,16 @@ export const RootStack: NavigatorType = () => {
       })
   }, [])
 
+  useAddDeviceToken()
   useEffect(() => {
     const setupNotifications = async () => {
-      if (hasToken && client) {
-        const hasPermission = await hasNotificationPermission()
-        if (hasPermission) {
-          addDeviceToken(client)
-          messaging().onTokenRefresh(() => addDeviceToken(client))
-        }
+      const hasPermission = await hasNotificationPermission()
+      if (hasPermission) {
+        messaging().onTokenRefresh(useAddDeviceToken)
```

Comment:
Hooks need to run on every render, I dont believe they can be used in a useEffect which happens outside of a render cycle and only when its dependencies change.